### PR TITLE
Feature/refactoring

### LINKED ISSUE #130: Historical Base64 image
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Current historical base64 image is in the javascript application, could it please be configurable as an entity or the PNG put in the www folder so it can be swapped to images of other brands like:
image

### Expected behaviour

Image based on model and modern

### Possible Solutions

1. Use model & modern: false combination to present image for other inverters
2. provide a script include file for base64 image with true / false driver
3. externalise png to folder on HA so it can be replaced but not overwritten during updates.

### Mode

full

### Context / Reason

Just to make your card more cool - it's part of a lot of our lives!

### PR DESCRIPTION
**Pull request by Pho3niX90**
- There are some code refactoring happening, to try and reduce duplication, redundant checks. The goal is to reduce the overall size and line count.
- Fixed some status code from the solis
- Change common strings to interfaces and enums for type safety
- Lint warning fixes
- Move dev packages to dev, to reduce bundle size

## Related issues
Resolves #130 

## Motivation and Context
The bulk of the PR is related to reducing complexity, and reducing the dom size.

## How has this been tested
I have been running this for the last couple days. 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have updated the version in `package.json` following [semver](https://semver.org/)

### CODE REVIEW

**Comment by slipx06** [comment]
There seems to be a problem with the code. It might be related to toNum or Number.isNaN but the battery duration and runtime is not working. Can you check

image

File: src/index.ts [hunk 1]
```diff
@@ -721,44 +753,48 @@ export class SunsynkPowerFlowCard extends LitElement {
                                   class="${additional_aux_load === 2 ? '' : 'st12'}"/>
                             <text id="duration" x="132" y="352" class="${font !== true ? 'st14' : 'st4'} left-align"
```
Comment (slipx06) [inline src/index.ts:754]: need to debug why this is not working lines 754-773

Comment (slipx06) [reply src/index.ts:754]: OK the problem is 

let battery_energy = Number.isNaN(this.toNum(battenergy.state)) ? config.battery.energy : this.toNum(battenergy.state);

Comment (Pho3niX90) [reply src/index.ts:754]: Will quickly have a look, apologies, never saw it working on my end as I had a workaround for my lead acids, lithium will be here this week.

**Comment by slipx06** [comment]
Its not reading the config.battery.energy number so its not evaluating the line below as intended

`let battery_energy = Number.isNaN(this.toNum(battenergy.state)) ? config.battery.energy : this.toNum(battenergy.state);`

This works
`let battery_energy = isNaN(parseInt(battenergy.state)) ? config.battery.energy : this.toNum(battenergy.state);`

We will have the same problem with 
`let shutdown = Number.isNaN(shutdownsoc.state) ? config.battery.shutdown_soc : this.toNum(shutdownsoc.state);`

Maybe there is a better way of implementing the logic. I want it to read the value in the config unless a sensor is suppled

File: src/index.ts [hunk 2]
```diff
@@ -345,8 +360,8 @@ export class SunsynkPowerFlowCard extends LitElement {
         };
 
         const shutdownsoc = this.hass.states[config.battery.shutdown_soc] || {state: ''};
-        let shutdown = isNaN(parseInt(shutdownsoc.state)) ? config.battery.shutdown_soc : parseInt(shutdownsoc.state);
-        let inverter_prog: inverterProg = {
+        let shutdown = Number.isNaN(shutdownsoc.state) ? config.battery.shutdown_soc : this.toNum(shutdownsoc.state);
```
Comment (slipx06) [inline src/index.ts:363]: Check

File: src/index.ts [hunk 3]
```diff
@@ -436,14 +452,14 @@ export class SunsynkPowerFlowCard extends LitElement {
         let duration = "";
 
         const battenergy = this.hass.states[config.battery.energy] || {state: ''};
-        let battery_energy = isNaN(parseInt(battenergy.state)) ? config.battery.energy : parseInt(battenergy.state);
+        let battery_energy = Number.isNaN(this.toNum(battenergy.state)) ? config.battery.energy : this.toNum(battenergy.state);
```
Comment (slipx06) [inline src/index.ts:455]: Check

**Comment by Pho3niX90** [comment]
Okay I made a commit now, I did some quick tests and seems to resolve it. If you can just confirm your side as well

**Comment by slipx06** [comment]
Perfect. Everything looks good. Just finishing some checks. Then I'll merge. Thanks for all your input.

**Pull request merged**